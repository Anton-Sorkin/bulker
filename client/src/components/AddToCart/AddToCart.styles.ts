import styled from "@emotion/styled";

export const AddToCardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 10px;
  height: 120px;
  width: 95vw;
  background-color: white;
  border-radius: 8px 0px 0px 8px;
  justify-content: center;
  align-items: center;
  position: relative;
  @media (min-width: 768px) {
    height: 140px;
  }
  & > button {
    background-color: #ff9b85;
    border: none;
    width: 70%;
    height: 50px;
    color: white;
    border-radius: 8px;
    cursor: pointer;
    @media (min-width: 768px) {
      width: 50%;
    }
  }
  & > p {
    font-size: 12px;
    font-weight: 300;
    color: #232323;
    position: absolute;
    top: 75px;
    @media (min-width: 768px) {
      top: 90px;
    }
  }
`;
