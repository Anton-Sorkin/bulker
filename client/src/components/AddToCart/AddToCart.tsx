import { AddToCardWrapper } from "./AddToCart.styles";

type AddToCartProps = {
  isSetToGroceryList: boolean;
  addToGroceryList: () => void;
  toggleConfirmedModal: () => void;
};

const AddToCart = ({
  isSetToGroceryList,
  addToGroceryList,
  toggleConfirmedModal,
}: AddToCartProps) => {
  return (
    <AddToCardWrapper>
      {isSetToGroceryList === false && (
        <button onClick={addToGroceryList}>Add to Grocery List</button>
      )}

      {isSetToGroceryList === true && (
        <>
          <button onClick={toggleConfirmedModal}>
            Add Another Grocery List
          </button>
          <p>it has now been added to your cart!</p>
        </>
      )}
    </AddToCardWrapper>
  );
};

export default AddToCart;
