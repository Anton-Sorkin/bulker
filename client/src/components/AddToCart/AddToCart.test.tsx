import React from "react";
import { render, fireEvent } from "@testing-library/react";
import AddToCart from "./AddToCart";
import { AddToCartProps } from "./AddToCart";

describe("AddToCart component", () => {
  test("should render correctly with isSetToGroceryList=false", () => {
    const addToGroceryList = jest.fn();
    const toggleConfirmedModal = jest.fn();
    const props: AddToCartProps = {
      isSetToGroceryList: false,
      addToGroceryList,
      toggleConfirmedModal,
    };
    const { getByText, asFragment } = render(<AddToCart {...props} />);
    const button = getByText("Add to Grocery List");
    fireEvent.click(button);
    expect(addToGroceryList).toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render correctly with isSetToGroceryList=true", () => {
    const addToGroceryList = jest.fn();
    const toggleConfirmedModal = jest.fn();
    const props: AddToCartProps = {
      isSetToGroceryList: true,
      addToGroceryList,
      toggleConfirmedModal,
    };
    const { getByText, asFragment } = render(<AddToCart {...props} />);
    const button = getByText("Add Another Grocery List");
    fireEvent.click(button);
    expect(toggleConfirmedModal).toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });
});
