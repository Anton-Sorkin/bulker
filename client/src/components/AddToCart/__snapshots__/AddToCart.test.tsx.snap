// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCart component should render correctly with isSetToGroceryList=false 1`] = `
<DocumentFragment>
  <div
    class="css-j0hugt"
  >
    <button>
      Add to Grocery List
    </button>
  </div>
</DocumentFragment>
`;

exports[`AddToCart component should render correctly with isSetToGroceryList=true 1`] = `
<DocumentFragment>
  <div
    class="css-j0hugt"
  >
    <button>
      Add Another Grocery List
    </button>
    <p>
      it has now been added to your cart!
    </p>
  </div>
</DocumentFragment>
`;
